source/common/http/http1/codec_impl.cc

```diff
@@ -575,7 +594,10 @@ StreamEncoder& ClientConnectionImpl::newStream(StreamDecoder& response_decoder)
   if (resetStreamCalled()) {
     throw CodecClientException("cannot create new streams after calling reset");
   }
-
+  // read-enable any underlying connection disabled by flow control on the prior request.
+  while (!connection_.readEnabled()) {
```

Comment:
As I understand it (and please correct me if I'm wrong) there can be only one user of an HTTP connection at a time (no true pipelining support).  I thought on newStream no one is using it yet, so it seemed like the best place to reset state.

What we need is to ensure that if a upstream connection overflows the downstream buffer, gets told "stop reading", disassociates (stream end), and the connection is reused, that we read-enable.   I don't *think* the other direction can happen: if upstream gets overrun either it should fully drain or the downstream connection can't be reused, but if we can have a fix in the codec I'm still happier.

If you can suggest somewhere else in the connection pool we can make sure that upstream connections can get unblocked before they're reused, I'd be happy to move this.